staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go

```diff
@@ -33,6 +33,11 @@ type Error struct {
 	Field    string
 	BadValue interface{}
 	Detail   string
+
+	// Origin indicates the source of this error.  It is used in testing to
+	// ensure that the expected errors are returned without matching on detail
+	// strings.
+	Origin string
```

Comment:
Would it make sense for `Origin` to be an enum?  Might help users/test-writers know what the valid “origins” are.

Reply:
I wanted to leave it open-ended so very custom validation can do it's own thing.  Like there's not a generic validator for ServiceAccount names, so that code might set it to "ServiceAccountName".

What I wonder most is whther this is specific ENOUGH to do correlation.